crates/wasmtime/src/runtime/vm/instance/allocator.rs

```diff
@@ -92,7 +92,14 @@ pub struct InstanceAllocationRequest<'a> {
 /// InstanceAllocationRequest, rather than on a &mut InstanceAllocationRequest
 /// itself, because several use-sites require a split mut borrow on the
 /// InstanceAllocationRequest.
-pub struct StorePtr(Option<NonNull<dyn VMStore>>);
+pub struct StorePtr(Option<crate::vm::SendSyncPtr<dyn VMStore>>);
```

Comment:
Mind leaving this as `NonNull` to avoid the cognitive overhead of using `SendSyncPtr`?

Reply:
Oh huh, I guess I lost this in a rebase or something, because I definitely undid this change at one point